gp-includes/misc.php

```diff
@@ -331,3 +331,40 @@ function gp_gmt_strtotime( $string ) {
 
 	return $time;
 }
+
+/**
+ * Determines the format to use based on the selected format type or by auto detection based on the file name.
+ *
+ * Used during import of translations and originals.
+ *
+ * @param string $selected_format The format that the user selected on the import page.
+ * @param string $filename The filname that was uploaded by the user.
+ * @return object A GP_Format child object.
+ */
+function gp_get_import_file_format( $selected_format, $filename ) {
+	$format = gp_array_get( GP::$formats, $selected_format, null );
+
+	if ( ! $format ) {
+		$extension = pathinfo( $filename, PATHINFO_EXTENSION );
+		
+		foreach( GP::$formats as $format_entry ) {
+			if( $extension == $format_entry->extension ) {
```

Comment:
The first line can actually be a public method of  `GP_Format`

```php
abstract class GP_Format {
	public $extension = '';
	public $alt_extensions = array();

	/**
	 * Gets the list of supported file extensions.
	 *
	 * @since 1.1.0
	 *
	 * @return array Supported file extensions.
	 */
	public function get_file_extensions() {
			return array_merge( array( $format_entry->extension ), $format_entry->alt_extensions );
	}
}
```